Marlin/src/lcd/menu/menu.cpp

```diff
@@ -168,8 +168,6 @@ void MenuEditItemBase::goto_edit_screen(
  */
 void MarlinUI::goto_screen(screenFunc_t screen, const uint16_t encoder/*=0*/, const uint8_t top/*=0*/, const uint8_t items/*=0*/) {
   if (currentScreen != screen) {
-    thermalManager.set_menu_cold_override(false);
```

Comment:
This is here to make sure the flag gets set to `false` whenever the move screen is exited (so, whenever the screen is changed), otherwise the flag will be stuck in the `true` state. As long as `set_menu_cold_override` is called after `_goto_menu_move_distance_e` the desired result should be achieved (assuming there are no sub-screens).

The call to `set_menu_cold_override(false)` that's being added to `menu_move` (below) won't do the trick. It only resets the flag to false in the event the user enters or returns to the Move menu, which is not guaranteed. It doesn't handle timeout to status screen or the case of some incoming G-code command changing the screen. Setting it to `false` here in `goto_screen` is expected to be totally reliable.

So, we need a better understanding why the current pattern is failing. Where else is the flag getting set to `false`? Is `goto_screen` called a second time by some quirk? What does investigation with `SERIAL_ECHO` tell us?